Add SOURCE_DATE_EPOCH support for build date

kpcyrd opened the pull request:
This uses the timestamp in [`$SOURCE_DATE_EPOCH`][1] if set and falls back to the current behavior otherwise. It tries `date -d "@..."` first and falls back to `date -r "..."` to support BSD date(1) too.

I'm not sure if I'm opening this PR to the right branch, let me know and I can rebase if needed.

Thanks!

[1]: https://reproducible-builds.org/docs/source-date-epoch/

syzop commented:
Sounds good. I assume you tested it on both Linux and FreeBSD indeed. If you could rebase for unreal60_dev that would be great.

kpcyrd committed:
Add SOURCE_DATE_EPOCH support for build date

kpcyrd commented:
I've rebased and updated the PR to unreal60_dev. The code is copied from the FreeBSD compatible reproducible builds example but I've only tested this on Linux because I don't have a FreeBSD system to test this with.

kpcyrd commented:
Thanks!

syzop commented:
I took the chance. Seems to work on all Linux and FreeBSD (12) buildbots (not public).

Thanks! :)
